### DISCUSSION TITLE
No install

### DISCUSSION THREAD

**Discussion by ChrisL79** [magicbug/Cloudlog#2776] [General]
Install ends in this error:

Error Number: 1146

Table 'db_cloudlog.users' doesn't exist

ALTER TABLE `users` ADD user_clublog_name VARCHAR(255) DEFAULT NULL, ADD user_clublog_password VARCHAR(255) DEFAULT NULL, ADD user_clublog_callsign VARCHAR(255) DEFAULT NULL

Filename: migrations/018_clubloguserfields.php

Line Number: 14

Any ideas to what the problem is? Tired of fighting this issue.

**Comment by HB9HIL** [comment]
Did you create the database before you run the installer? You have to create the database by yourself before. The installer just fills the base data to the database

**Comment by ChrisL79** [comment]
Yes. The solution was to wait a few minutes on that error screen and Cloudlog install screen was then shown shortly afterwards. 

I tried this on varioius versions of Ubuntu server with the same error.

A friend also tried on a separate server with the same error. 

Just wait and eventually it will install.

Must be a bug in the latest version.

**Reply by HB9HIL** [reply]
I'll check

**Reply by bgritzmacher** [reply]
@HB9HIL Looking through the install.php file, I'm not certain that it ensures that `Database\create_tables` completes properly through its use of [mysqli::multi_query](https://www.php.net/manual/en/mysqli.multi-query.php).

https://github.com/magicbug/Cloudlog/blob/ff3738e2bff6a775180c489f50b77f33ece4bbdf/install/includes/database_class.php#L25-L44

Reading the documentation for `mysqli::multi_query`:

> Queries are sent asynchronously in a single call to the database, but the database processes them sequentially. mysqli_multi_query() waits for the first query to complete before returning control to PHP. The MySQL server will then process the next query in the sequence. Once the next result is ready, MySQL will wait for the next execution of [mysqli_next_result()](https://www.php.net/manual/en/mysqli.next-result.php) from PHP.

That sounds like it doesn't block the rest of the script execution after the first instruction in the install.sql file, allowing it to return `true` (not that there's any way for it to return `false`, by the way,) and continue on as if everything is fine, no?

If that's the case, it would certainly explain why the problem fixes itself after a minute or so - because the MySQL server is still processing the commands for creating the tables!

As per the PHP documentation, they have this recommendation:

> It is recommended to use [do-while](https://www.php.net/manual/en/control-structures.do.while.php) to process multiple queries. The connection will be busy until all queries have completed and their results are fetched to PHP. No other statement can be issued on the same connection until all queries are processed. To proceed to the next query in the sequence, use [mysqli_next_result()](https://www.php.net/manual/en/mysqli.next-result.php). If the next result is not ready yet, mysqli will wait for the response from the MySQL server. To check if there are more results, use [mysqli_more_results()](https://www.php.net/manual/en/mysqli.more-results.php).

This would make sure that the install script does not proceed until after the MySQL has processed all of the commands.

**Comment by HB9HIL** [comment]
I'm really sorry @ChrisL79 but I can't reproduce this problem here. Tested with Ubuntu 22.04 and Debian 12. Site is loading normally and after everything is done the login screen is shown.

Can you give me more information about the hardware you are using? If you want to can E-Mail me directly (Mail is on QRZ.com). This seems to be a local problem. For me it looks like the hardware is not fast enough to do the job before php timeout kicks in. 

@bgritzmacher Thanks for reasearching this. Maybe you are right. But in fact we rarely receive a problem in this matter and because I have another install process on my personal roadmap I won't touch the existing install script. :)

**Comment by ChrisL79** [comment]
It installed eventually, so all is good. This was tested on Proxmox with 32GB allocated and i5. Ubuntu 18.4 and 22.4.3 Also tested natively on Ryzen 7with 16gb (separate server) 18.4 and 22.4.3. Also tested natively on a friends server and same issue was produced with 22.4.3.

Many thanks to all that looked into it. It was much appreciated.

Bottom line is i have a functioning Cloudlog.

Now to just sort out why CloudlogCat will not install on Windows10. Active X errors.

**Reply by HB9HIL** [reply]
okay so seems like the machine has enough power. next step would be check of php.ini. But if your Cloudlog is running now everything seems fine :)

If this issue is not longer relevant for you please close it 👍 thx :)

**Comment by ChrisL79** [comment]
Sorry, i thought i made it clear in my post that it will install fine, you just have to wait. I spent 2 days of trying differemt configurations where all i needed to do was wait.

At least this is here for anyone else that may have same issue in the future.

I will mark this issue closed.

Thanks again.
